apis/cluster/v1beta1/membercluster_types.go

```diff
@@ -49,6 +49,14 @@ type MemberClusterSpec struct {
 	HeartbeatPeriodSeconds int32 `json:"heartbeatPeriodSeconds,omitempty"`
 }
 
+// Metric is the observed value of a non-resource metric.
+type Metric struct {
+	// The name of the metric; it should be a valid Kubernetes label name.
```

Comment:
do we need a metrics type? for example, static vs dynamic? integer vs fraction?

Reply:
The static vs dynamic distinction might make less sense IMO as we don't really handle them differently (assuming that static metrics = metrics that don't change very often?). 

As for the int vs float distinction, Kubernetes does not recommend using floats directly in the API (as different programming languages support them differently); for obvious reasons, we will always need float values, so here all values are serialized into a string as recommended, which should work just fine for both ints and floats.